internal/thelma/utils/schedule/schedule.go

```diff
@@ -0,0 +1,44 @@
+package schedule
+
+import "time"
+
+// CheckDailyScheduleMatch determines if the `schedule` time's daily repetition happened between `since` and `now`.
+// We assume that `since` is before `now`.
+//
+// An important edge case here is if midnight falls in between `since` and `now`. We want to convert `schedule` to the
+// correct day, but there's now two possible candidates.
+//
+// We handle this case by always initially converting `schedule` to `now`'s day. If `schedule` is then ahead of `now`,
```

Comment:
Would working with unix timestamps work as another way to solve this without hitting the edge case?

Reply:
Although that conversion may be more trouble than it's worth

Reply:
Possibly--we'd still need to find the correct daily repetition of the schedule time, and in my mind this was easier and potentially more correct than modulo by the number of seconds in a day. Go is doing the .Before and .After using the unix timestamp, working in terms of human-readable dates just seemed like an easier way to get there.